examples/armv4t/gdb/catch_syscalls.rs

```diff
@@ -0,0 +1,20 @@
+use gdbstub::target;
+use gdbstub::target::ext::catch_syscalls::SyscallNumbers;
+
+use crate::gdb::Emu;
+
+// This implementation is for illustrative purposes only. If the target doesn't
+// support syscalls then there is no need to implement this extension
+
+impl target::ext::catch_syscalls::CatchSyscalls for Emu {
+    fn enable_catch_syscalls(
+        &mut self,
+        _filter: Option<SyscallNumbers<u32>>,
+    ) -> target::TargetResult<(), Self> {
+        Ok(())
```

Comment:
ah, one last thing: even though the example code never performs any syscalls, it should still be possible to observe that `gdbstub` has received the request to catch syscalls.

Could you add some `eprintln!` calls to the bodies of these functions, and validate that they are being called correctly? e.g:

```rust
match filter {
    Some(numbers) => eprintln!("Enabled catching syscalls: {:?}", numbers.collect::<Vec<u32>>()),
    None => eprintln!("Enabled catching all syscalls"),
};

eprintln!("Disabled catching syscalls")
```